src/hdmf/container.py

```diff
@@ -51,7 +51,11 @@ def add_child(self, **kwargs):
         if child is not None:
             self.__children.append(child)
```

Comment:
I think you want to move this under the conditional as well

Reply:
This would no longer be true:
```python
parent_obj = Container('obj1')
child_obj = Container('obj2', parent_obj)
self.assertIs(parent_obj.children[0], child_obj)
```
because `child_obj.parent` gets set before `child_obj.parent.add_child(child_obj)` gets called in lines 20-22 above. 
I had added lines 21-22 because I thought the parent should know the objects that call it its parent, but given how the code works, this doesn't seem to be important, so just in case that breaks something as well, I'll revert it.